framework/src/media/Decoder.cpp

```diff
@@ -51,6 +47,50 @@ Decoder::~Decoder()
 #endif
 }
 
+std::shared_ptr<Decoder> Decoder::create(audio_type_t audioType, unsigned short channels, unsigned int sampleRate)
+{
+#ifdef CONFIG_AUDIO_CODEC
+	medvdbg("(%d,%d,%d)\n", audioType, channels, sampleRate);
+	
+	std::shared_ptr<Decoder> instance(new Decoder(audioType, channels, sampleRate));
+	if (instance) {
+		if (instance->init()) {
+			return instance;
+		}
+	}
+
+	meddbg("Error! (%d,%d,%d), failed!\n", audioType, channels, sampleRate);
+	return nullptr;
+#else
+	return nullptr;
+#endif
+}
+
+bool Decoder::init(void)
+{
+#ifdef CONFIG_AUDIO_CODEC
+	memset(&mDecoder, 0, sizeof(audio_decoder_t));
+	if (audio_decoder_init(&mDecoder, CONFIG_AUDIO_CODEC_RINGBUFFER_SIZE) != AUDIO_DECODER_OK) {
+		meddbg("Error! audio_decoder_init failed!\n");
+		return false;
+	}
+
+	mDecoder.audio_type = mAudioType;
+	if (mDecoder.audio_type != AUDIO_TYPE_UNKNOWN) {
```

Comment:
Actually we will provides header parser (#2059), I think we can return error for unknown case. if you agree with me, how about below?
```
if (type == UNKNOWN) {
    return false;
}
if (mConfig(audioType) {
     dbg
     return false;
}
return true;
```